core-java-modules/core-java-regex/src/test/java/com/baeldung/regex/findnumbers/FindNumbersUnitTest.java

```diff
@@ -0,0 +1,83 @@
+package com.baeldung.regex.findnumbers;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.contains;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.junit.jupiter.api.Test;
+
+/**
+ * Unit Test to find Integers and Decimal Numbers in a String
+ */
+class FindNumbersUnitTest {
+
+    private static List<String> findIntegers(String stringToSearch) {
+        Pattern integerPattern = Pattern.compile("-?\\d+");
+        Matcher matcher = integerPattern.matcher(stringToSearch);
+
+        List<String> integerList = new ArrayList<>();
+        while (matcher.find()) {
+            integerList.add(matcher.group());
+        }
+
+        return integerList;
+    }
+
+    private static List<String> findDecimalNums(String stringToSearch) {
+        Pattern decimalNumPattern = Pattern.compile("-?\\d+(\\.\\d+)?");
+        Matcher matcher = decimalNumPattern.matcher(stringToSearch);
+
+        List<String> decimalNumList = new ArrayList<>();
+        while (matcher.find()) {
+            decimalNumList.add(matcher.group());
+        }
+
+        return decimalNumList;
+    }
+
+    @Test
+    void givenStrOfAllDigits_whenRegexMatchByInteger_thenWholeStrMatchedAsOneInteger() {     
+        List<String> integersFound = findIntegers("970987678607608");
+
+        assertThat(integersFound, contains("970987678607608"));
+    }
+
+    @Test
+    void givenStrWithIntegersSepByPeriods_whenRegexMatchByInteger_thenExpectedIntegersFound() {
+        List<String> integersFound = findIntegers("3453..5.-23532...32432.-2363.3454......345.-34.");
+
+        assertThat(integersFound, contains("3453", "5", "-23532", "32432", "-2363", "3454", "345", "-34"));
+    }
+
+    @Test
+    void givenStrWithIntegersSepByNonDigitsInclPeriods_whenRegexMatchByInteger_thenExpectedIntegersFound() {
+        List<String> integersFound = findIntegers("646lkl~4-53l-k34.fdsf.-ds-35.45f9wg3868*()(k;-95786fsd79-86");
+
+        assertThat(integersFound, contains("646", "4", "-53", "34", "-35", "45", "9", "3868", "-95786", "79", "-86"));
+    }
+    
+    @Test
+    void givenStrOfAllDigits_whenRegexMatchByDecimalNumber_thenWholeStrMatchedAsOneDecimalNumber() {
+        List<String> decimalNumsFound = findDecimalNums("970987678607608");
+        
+        assertThat(decimalNumsFound, contains("970987678607608"));
```

Comment:
MINOR - hamcrest `assertThat` is nowhere near as good as `AssertJ`. However, both are valid. I'd recommend considering AssertJ for all future assertions in all aspects of Java development :)

Reply:
contains and containsExactly in assertJ do what they should